test/intl402/Temporal/PlainDate/prototype/since/wrapping-at-end-of-month-persian.js

```diff
@@ -0,0 +1,101 @@
+    );
+  }
+}
+
+// Difference between end of longer month to end of not-immediately-following
+// shorter month
+{
+  const end = Temporal.PlainDate.from({ year: 1970, monthCode: "M08", day: 30, calendar });
+  for (const largestUnit of ["years", "months"]) {
+    TemporalHelpers.assertDuration(
+      Temporal.PlainDate.from({ year: 1970, monthCode: "M06", day: 30, calendar }).since(end, { largestUnit }),
+      0, -2, 0, 0, 0, 0, 0, 0, 0, 0,
+      `Shahrivar 30th to Aban 30th is 2 months (${largestUnit})`
+    );
+    TemporalHelpers.assertDuration(
+      Temporal.PlainDate.from({ year: 1970, monthCode: "M06", day: 31, calendar }).since(end, { largestUnit }),
+      0, -1, 0, -30, 0, 0, 0, 0, 0, 0,
+      `Shahrivar 30th to Aban 29th is 1 month 30 days, not 2 months (${largestUnit})`
+    );
+  }
+}
+
+// Difference between end of longer month in one year to shorter month in
+// later year
+{
+  const end = Temporal.PlainDate.from({ year: 1971, monthCode: "M12", day: 29, calendar });
+  TemporalHelpers.assertDuration(
+    Temporal.PlainDate.from({ year: 1970, monthCode: "M06", day: 29, calendar }).since(end, { largestUnit: "months" }),
+    0, -18, 0, 0, 0, 0, 0, 0, 0, 0,
+    "Shahrivar 29th 1970 to Esfand 29th 1971 is 18 months"
+  );
+  TemporalHelpers.assertDuration(
+    Temporal.PlainDate.from({ year: 1970, monthCode: "M06", day: 29, calendar }).since(end, { largestUnit: "years" }),
+    -1, -6, 0, 0, 0, 0, 0, 0, 0, 0,
+    "Shahrivar 29th 1970 to Esfand 29th 1971 is 1 year, 6 months"
+  );
+  TemporalHelpers.assertDuration(
+    Temporal.PlainDate.from({ year: 1970, monthCode: "M06", day: 30, calendar }).since(end, { largestUnit: "months" }),
+    0, -17, 0, -29, 0, 0, 0, 0, 0, 0,
+    "Shahrivar 30th 1970 to Esfand 29th 1971 is 17 months, 29 days, not 18 months"
+  );
+  TemporalHelpers.assertDuration(
+    Temporal.PlainDate.from({ year: 1970, monthCode: "M06", day: 30, calendar }).since(end, { largestUnit: "years" }),
+    -1, -5, 0, -29, 0, 0, 0, 0, 0, 0,
+    "Shahrivar 30th 1970 to Esfand 29th 1971 is 1 year, 5 months, 29 days, not 1 year 6 months"
+  );
+}
+
+// Difference where months passes through a month that's the same length or
+// shorter than either the start or end month
+{
+  TemporalHelpers.assertDuration(
+    Temporal.PlainDate.from({ year: 1970, monthCode: "M06", day: 31, calendar })
+      .since(Temporal.PlainDate.from({ year: 1971, monthCode: "M01", day: 30, calendar }), { largestUnit: "months" }),
+    0, -6, 0, -30, 0, 0, 0, 0, 0, 0,
+    "Shahrivar 31st 1970 to Farvardin 30th 1971 is 6 months 30 days, not 209 days"
+  );
+  TemporalHelpers.assertDuration(
+    Temporal.PlainDate.from({ year: 1970, monthCode: "M06", day: 31, calendar })
+      .since(Temporal.PlainDate.from({ year: 1972, monthCode: "M01", day: 30, calendar }), { largestUnit: "years" }),
+    -1, -6, 0, -30, 0, 0, 0, 0, 0, 0,
+    "Shahrivar 31st 1970 to Farvardin 30th 1972 is 1 year, 6 months, 30 days"
+  );
+}
```

Comment:
Here I think it'd be best to use a common year and start in the end of Bahman (M11), so that we go from a short month, pass through a shorter month, and end in a long month. This most closely matches the situation I was trying to test for in the Gregorian calendar, where if the algorithm was implemented incorrectly we'd get something like "59 days" for that input.